deeppavlov/core/commands/train.py

```diff
@@ -341,7 +342,7 @@ def improved(score, best):
         tb_valid_writer = tf.summary.FileWriter(str(tb_log_dir / 'valid_log'))
 
     # validate first (important if model is pre-trained)
-    if train_config['val_every_n_epochs'] > 0 and epochs % train_config['val_every_n_epochs'] == 0:
+    if train_config['val_every_n_epochs'] > 0 or train_config['val_every_n_batches'] > 0:
         report = _test_model(model, metrics_functions, iterator,
```

Comment:
if `val_everyn_epochs > 1` model will be though tested every epoch.

Reply:
This `if` is not run in any `while` block, it will be executed only once